hibernate-core/src/main/java/org/hibernate/boot/model/naming/NamingHelper.java

```diff
@@ -174,4 +138,13 @@ public String hashedName(String s) {
 			throw new HibernateException( "Unable to generate a hashed name!", e );
 		}
 	}
+
+	private String generateColumnsHash(String prefix, StringBuilder sb, Identifier[] columnNames) {
+		// Ensure a consistent ordering of columns, regardless of the order
+		// they were bound.
+		Arrays.stream( columnNames )
+				.sorted()
+				.forEachOrdered( columnName -> sb.append( "column`" ).append( columnName ).append( '`' ) );
```

Comment:
The above code snippet might be slower than traditional approach. It would be great to conduct some JMH benchmark to justify the stream usage.